android/engine/src/main/java/org/smartregister/fhircore/engine/pdf/HtmlPopulator.kt

```diff
@@ -16,32 +16,53 @@
 
 package org.smartregister.fhircore.engine.pdf
 
+import java.util.Date
 import java.util.regex.Matcher
 import java.util.regex.Pattern
 import org.hl7.fhir.r4.model.BaseDateTimeType
 import org.hl7.fhir.r4.model.QuestionnaireResponse
+import org.hl7.fhir.r4.model.QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent
 import org.smartregister.fhircore.engine.util.extension.allItems
+import org.smartregister.fhircore.engine.util.extension.extractLogicalIdUuid
 import org.smartregister.fhircore.engine.util.extension.formatDate
 import org.smartregister.fhircore.engine.util.extension.makeItReadable
 import org.smartregister.fhircore.engine.util.extension.valueToString
 
 /**
  * HtmlPopulator class is responsible for processing an HTML template by replacing custom tags with
- * data from a QuestionnaireResponse. The class uses various regex patterns to find and replace
- * custom tags such as @is-not-empty, @answer-as-list, @answer, @submitted-date, and @contains.
+ * data from QuestionnaireResponses. The class uses various regex patterns to find and replace
+ * custom tags such as @is-not-empty, @answer-as-list, @answer, @submitted-date, @contains,
+ * and @is-questionnaire-submitted.
  *
- * @property questionnaireResponse The QuestionnaireResponse object containing data for replacement.
+ * @property questionnaireResponses The QuestionnaireResponses object containing data for
+ *   replacement.
  */
 class HtmlPopulator(
-  private val questionnaireResponse: QuestionnaireResponse,
+  questionnaireResponses: List<QuestionnaireResponse>,
 ) {
+  // Store answers as key-value pairs with link-id as key
+  private val answerMap: MutableMap<String, List<QuestionnaireResponseItemAnswerComponent>> =
+    mutableMapOf()
 
-  // Map to store questionnaire response items keyed by their linkId
-  private val questionnaireResponseItemMap =
-    questionnaireResponse.allItems.associateBy(
-      keySelector = { it.linkId },
-      valueTransform = { it.answer },
-    )
+  // Store submitted-date of a questionnaire response with link-id as key
+  private val submittedDateMap: MutableMap<String, Date> = mutableMapOf()
+
+  // Store questionnaire-id related to each questionnaire-response
+  private val questionnaireIds: MutableList<String> = mutableListOf()
+
+  init {
+    questionnaireResponses.forEach { questionnaireResponse ->
+      val questionnaireId = questionnaireResponse.questionnaire.extractLogicalIdUuid()
+      val answerMap =
+        questionnaireResponse.allItems.associateBy(
+          keySelector = { "$questionnaireId/${it.linkId}" },
+          valueTransform = { it.answer },
+        )
```

Comment:
You can convert this to a `Map` instead of `MutableMap`, declare it as a `lateinit` property, and, then initialize it in the `init` method. 

```kotlin
 private lateinit var answerMap: Map<String, List<QuestionnaireResponseItemAnswerComponent>>

init {
      this.answerMap =
        questionnaireResponse.allItems.associateBy(
          keySelector = { "$questionnaireId/${it.linkId}" },
          valueTransform = { it.answer },
        )
}
```